src/ResourceManager/DataFactories/Commands.DataFactoryV2/Runs/StopAzureDataFactoryPipelineRunCommand.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.DataFactoryV2.Models;
+using Microsoft.Azure.Commands.DataFactoryV2.Properties;
+using System;
+
+namespace Microsoft.Azure.Commands.DataFactoryV2
+{
+    [Cmdlet(VerbsLifecycle.Stop, Constants.PipelineRun, DefaultParameterSetName = ParameterSetNames.ByFactoryName, SupportsShouldProcess = true), OutputType(typeof(bool))]
+    public class StopAzureDataFactoryPipelineRunCommand : DataFactoryContextBaseCmdlet
+    {
+        [Parameter(ParameterSetName = ParameterSetNames.ByInputObject, Position = 0, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = Constants.HelpPipelineRunId)]
+        [ValidateNotNullOrEmpty]
+        public PSPipelineRun InputObject { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryName, Position = 2, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpPipelineRunId)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByFactoryObject, Position = 1, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = Constants.HelpPipelineRunId)]
+        [ValidateNotNullOrEmpty]
+        public string PipelineRunId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.HelpDontAskConfirmation)]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        [EnvironmentPermission(SecurityAction.Demand, Unrestricted = true)]
+        public override void ExecuteCmdlet()
+        {
+            ByFactoryObject(DataFactory);
+            if (ParameterSetName.Equals(ParameterSetNames.ByInputObject, StringComparison.OrdinalIgnoreCase))
```

Comment:
You might also move ByInputObject from DataFactoryContextActionBaseCmdlet to DataFactoryContextBaseCmdlet so you can use it here.  Alternatively, is there a reason you aren't inheriting from DataFactoryContextActionBaseCmdlet?

Reply:
Not this one because it is also assigning the pipeline run id in case by input object.
ActionBase also has ResourceId parameter which can't be added to this cmdlet as the pipelineRun is not an azure resource.

Reply:
That makes sense, got it.